### PR TITLE
Vs2022 cpp20

### PR DESCRIPTION
**Pull request by Blackvals**
- Polished Jenkins pipeline merging both into Build.pipeline and orchestrating the build process through Conan (who in turn uses CMake)
- Removed sqleet code and gather it as a conan dependency
- Updated DbSQLiteAdapter to comply with latest version of DbAdapterInterface and DbAdapterTestUtilities. Main changes are:
    - VS2022 and C++20
    - Removing boost in favor of standard
        - boost::optional by std::optional
        - boost::posix_time and boost::gregorian by DateTimeType from the interface which encapsulates std::chrono timepoint (system_clock, configurable precision, default microsec).

### CODE REVIEW

File: src/DbSQLiteAdapter/CMakeLists.txt [hunk 1]
```diff
@@ -2,9 +2,10 @@ cmake_minimum_required(VERSION 3.2)
 
 # Find external dependencides
 find_package(DbAdapterInterface)
+find_package(sqleet)
 
 # Configure preprocessor definitions
-add_compile_options(-DBOOST_ALL_DYN_LINK -DSQLITE_HAS_CODEC)
```
Comment (Thymir) [inline src/DbSQLiteAdapter/CMakeLists.txt:7]: test\DbSQLiteAdapterTest\CMakeLists.txt
and
test_package\CMakeLists.txt

still have -DBOOST_ALL_DYN_LINK which could be deleted

File: conanfile.py [hunk 2]
```diff
@@ -13,11 +13,11 @@ class DbSQLiteAdapterConan(ConanFile):
 	exports_sources = "*", "!build*", "!*.yml", "!*.md", "!*.in", "!ci", "!.gitattributes", "!.gitignore", "!LICENSE"
```
Comment (Thymir) [inline conanfile.py:14]: options of test_package/conanfile.py should be deleted

**Review by Thymir — APPROVED** [review]
Agree with implementation. Great Job. 2 comments about tests

File: src/DbSQLiteAdapter/CMakeLists.txt (hunk 1 above)
Comment (Blackvals) [reply src/DbSQLiteAdapter/CMakeLists.txt:7]: You are right, corrected

File: conanfile.py (hunk 2 above)
Comment (Blackvals) [reply conanfile.py:14]: Right, corrected and tested locally

**Pull request merged**